indico/modules/rb_new/client/js/components/App.jsx

```diff
@@ -79,21 +79,20 @@ export default class App extends React.Component {
                     <div styleName="rb-content">
                         <Switch>
                             <Route exact path="/" render={() => <Redirect to="/book" />} />
-                            {!isInitializing && (
-                                <>
-                                    <Route path="/book" render={({location, match: {isExact}}) => (
-                                        filtersSet
-                                            ? (
-                                                <BookRoom location={location} />
-                                            ) : (
-                                                isExact ? <Landing /> : <Redirect to="/book" />
-                                            )
-                                    )} />
-                                    <Route path="/rooms" component={RoomList} />
-                                    <Route path="/blockings" component={BlockingList} />
-                                    <Route path="/calendar" component={Calendar} />
-                                </>
-                            )}
+                            <Route path="/book" render={({location, match: {isExact}}) => {
+                                if (isInitializing) {
+                                    return null;
+                                }
+                                return filtersSet
+                                    ? (
+                                        <BookRoom location={location} />
+                                    ) : (
+                                        isExact ? <Landing /> : <Redirect to="/book" />
+                                    );
+                            }} />
+                            <Route path="/rooms" component={!isInitializing ? RoomList : null} />
```

Comment:
What about something like this?
```
function ConditionalRoute({active, component, ...props}) {
    return <Route {...props} component={active ? component : null}>;
}
```

(also needs to handle the case of a render prop instead of component prop of course)

Reply:
Yeah, I had sth very similar to this (but had to remove it due to some problems which eventually turned out to be unrelated).